### PR TITLE
add sitemap_generator gem

### PR DESCRIPTION
**Pull request by nabeta**
This PR wil fix https://github.com/antleaf/nims-mdr-development/issues/480 .

### CODE REVIEW

**Review by anusharanganathan — COMMENTED** [review]
The workflow does not alter the visibility setting. So works can be in Draft and still be visible. (see example solr record [solr_dataset_response_json.txt](https://github.com/nims-dpfc/nims-hyrax/files/7184212/solr_dataset_response_json.txt)).  We would need to check workflow state is Deposited

The fedora state active or inactive state is used when records are deleted and tomb-stoned.

I will make changes to this PR.

Waiting on PR https://github.com/nims-dpfc/nims-hyrax/pull/411

**Comment by anusharanganathan** [comment]
> The workflow does not alter the visibility setting. So works can be in Draft and still be visible. (see example solr record [solr_dataset_response_json.txt](https://github.com/nims-dpfc/nims-hyrax/files/7184212/solr_dataset_response_json.txt)). We would need to check workflow state is Deposited
> 
> The fedora state active or inactive state is used when records are deleted and tomb-stoned.
> 
> I will make changes to this PR.
> 
> Waiting on PR #411

I got confused between the state in Solr and Fedora. The code is correct. Spotted another issue though.

File: hyrax/config/sitemap.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  #
+  # Add all articles:
+  #
+  #   Article.find_each do |article|
+  #     add article_path(article), :lastmod => article.updated_at
+  #   end
+  #
+  def build_each_doc(doc)
+    key = doc.fetch('has_model_ssim', []).first.constantize.model_name.singular_route_key
+    loc = Rails.application.routes.url_helpers.send(key + "_url", doc['id'], only_path: true)
```
Comment (nabeta) [inline hyrax/config/sitemap.rb:30]: @anusharanganathan Does the `<loc>` have to be an absolute URL?

Comment (nabeta) [reply hyrax/config/sitemap.rb:30]: Ah, it must be an absolute URL.
> This URL must begin with the protocol (such as http)
https://www.sitemaps.org/protocol.html

**Pull request merged**